tests/install_vm.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/python
+import argparse
+import os
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument(
+    "--distro",
+    dest="distro",
+    default="fedora",
+    choices=("fedora", "centos7"),
+    help="What type of distribution to install"
+)
+parser.add_argument(
+    "--domain",
+    dest="domain",
+    required=True,
+    help="What name should the new domain have."
+)
+parser.add_argument(
+    "--disk-dir",
+    dest="disk_dir",
+    default="/var/lib/libvirt/images/",
+    help="Location of the VM qcow file."
+)
+parser.add_argument(
+    "--ram",
+    dest="ram",
+    default=2048,
+    type=int,
+    help="Amount of RAM configured for the VM."
+)
+parser.add_argument(
+    "--cpu",
+    dest="cpu",
+    default=2,
+    type=int,
+    help="Number of CPU cores configured for the VM."
+)
+parser.add_argument(
+    "--dry",
+    dest="dry",
+    action="store_true",
+    help="Print command line instead of triggering command."
+)
+
+parser = parser.parse_args()
+data = parser.__dict__
+
+data['kickstart'] = "https://raw.githubusercontent.com/OpenSCAP/scap-security-guide/master/tests/kickstarts/rhel_centos_7.cfg"
+data['disk_path'] = os.path.join(parser.disk_dir, parser.domain) + ".qcow"
+
+if parser.distro == "fedora":
+    data['variant'] = "fedora27" # this is for support in RHEL7, where fedora28 is not known yet
+    data['url'] = "https://download.fedoraproject.org/pub/fedora/linux/releases/28/Everything/x86_64/os"
+elif parser.distro == "centos7":
+    data['variant'] = "centos7"
+    data['url'] = "http://mirror.centos.org/centos/7/os/x86_64"
+
+command = 'virt-install -n {domain} -r {ram} --vcpus={cpu} --os-variant={variant} --accelerate --disk path={disk_path},size=12 -x "inst.ks={kickstart}" --location {url}'.format(**data)
+if parser.dry:
+    print(command)
+else:
+    os.system(command)
```

Comment:
I suggest wrapping main in `def main():`, it's a nice paradigm that let's you import the python file without executing it. We use it everywhere else.

Reply:
I agree